tests/integration/s2n_sslyze_test.py

```diff
@@ -0,0 +1,140 @@
+#
+# Copyright 2017 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License").
+# You may not use this file except in compliance with the License.
+# A copy of the License is located at
+#
+#  http://aws.amazon.com/apache2.0
+#
+# or in the "license" file accompanying this file. This file is distributed
+# on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+# express or implied. See the License for the specific language governing
+# permissions and limitations under the License.
+#
+import argparse
+import os
+import sys
+import subprocess
+import itertools
+import multiprocessing
+import json
+import time
+from pprint import pprint
+from os import environ
+from multiprocessing.pool import ThreadPool
+from s2n_test_constants import *
+
+
+def cleanup_processes(*processes):
+    for p in processes:
+        p.kill()
+        p.wait()
+
+def run_sslyze_scan(endpoint, port, scan_output_location, enter_fips_mode=False):
+    """
+    Attempt to handshake against s2nd listening on `endpoint` and `port` using Openssl s_client
+
+    :param int endpoint: endpoint for s2nd to listen on
+    :param int port: port for s2nd to listen on
+
+    :param bool enter_fips_mode: True if s2nd should enter libcrypto's FIPS mode. Libcrypto must be built with a FIPS module to enter FIPS mode.
+
+    :return: 0 on successfully negotiation(s), -1 on failure
+    """
+    
+    s2nd_cmd = ["../../bin/s2nd"]
```

Comment:
I feel this stuff should be in a shared module that all of the test suites can use. Don't think we need to block this PR on it though.